src/js/components/JobForm.js

```diff
@@ -8,18 +8,33 @@ const METHODS_TO_BIND = [
   'validateForm'
 ];
 
+const SCHEDULE_FIELDS = ['cron', 'timezone', 'startingDeadlineSeconds'];
+
 class JobForm extends SchemaForm {
   constructor() {
-    super();
+    super(...arguments);
 
     METHODS_TO_BIND.forEach((method) => {
       this[method] = this[method].bind(this);
     });
   }
 
-  componentWillMount() {
+  handleFormChange(formData, eventObj) {
+    // Fire changes only on blur
+    // and on schedule runOnSchedule checkbox change to show/hide fields
+    let scheduleEnabledChange = formData.hasOwnProperty('cron') &&
+      eventObj.eventType === 'change' &&
+      eventObj.fieldName === 'runOnSchedule';
+    if (eventObj.eventType !== 'blur' && !scheduleEnabledChange) {
+      return;
+    }
+
+    this.validateForm();
+    // Update model before we update multipleDefinition
+    this.model = this.triggerTabFormSubmit();
```

Comment:
Would be nice if we find a way to not store the `model` on the form itself. But as this pattern is also found in the SchemaForm I'm ok if you don't address it.